--- a/configure.ac
+++ b/configure.ac
@@ -110,7 +110,7 @@
 	AS_HELP_STRING([--with-openssl], [handle PKCS7 signatures @<:@default=disabled@:>@]),
 	[], [with_openssl=no])
 AS_IF([test "x$with_openssl" != "xno"], [
-	PKG_CHECK_MODULES([openssl], [openssl >= 1.1.0])
+	PKG_CHECK_MODULES([openssl], [openssl >= 1.0.0])
 	AC_DEFINE([ENABLE_OPENSSL], [1], [Enable openssl for modinfo.])
 ], [
 	AC_MSG_NOTICE([openssl support not requested])
--- a/libkmod/libkmod-signature.c
+++ b/libkmod/libkmod-signature.c
@@ -187,7 +187,7 @@
 		return NULL;
 
 	d = X509_NAME_ENTRY_get_data(e);
-	str = (const char *)ASN1_STRING_get0_data(d);
+	str = (const char *)ASN1_STRING_data(d);
 
 	return str;
 }
@@ -246,7 +246,7 @@
 
 	CMS_SignerInfo_get0_algs(si, NULL, NULL, &dig_alg, &sig_alg);
 
-	sig_info->sig = (const char *)ASN1_STRING_get0_data(sig);
+	sig_info->sig = (const char *)ASN1_STRING_data(sig);
 	sig_info->sig_len = ASN1_STRING_length(sig);
 
 	sno_bn = ASN1_INTEGER_to_BN(sno, NULL);
